code/game/objects/items/miscellaneous.dm

```diff
@@ -63,6 +63,22 @@
 
 	new /obj/effect/DPtarget(get_turf(src), pod)
 
+//HoS gun beacon
+/obj/item/choice_beacon/hosgun
+	name = "Head of Security gun beacon"
+	desc = "Choose one of several experimental guns created by leading scientists at NT to put it through proper field testing."
+
+/obj/item/choice_beacon/hosgun/generate_display_names()
+	var/static/list/hosgun_item_list
+	if(!hosgun_item_list)
+		hosgun_item_list = list()
+		var/list/templist = typesof(/obj/item/gun/energy/e_gun/hos)
+		for(var/V in templist)
+			var/atom/A = V
+			hosgun_item_list[initial(A.name)] = A
+	return hosgun_item_list
```

Comment:
Bad code is bad.

Reply:
It's a copy and paste of the other choice beacons

Reply:
Doesn't make it less bad.

Reply:
I'm willing to put this PR to a vote but only after you've cleaned up this proc.